### DISCUSSION TITLE
Runtime xml config update

### DISCUSSION THREAD

**Discussion by patrikmnich** [tony19/logback-android#299] [Q&A]
Hi all,

I've been using logback in my application and I came to a point, where I'd like to know if there's a way to update logback.xml configuration in runtime. Here's the case:
I'm programming a payment application for terminal, but when we deploy it to production, we need to have control over configurations/files. This has been no issue for other configs using local app file folder etc., but logback specifically requires configuration to be stored in assets folder.
As you cannot operate with assets folder in this manner after application is built, I need to know if there is some other way to update xml configuration or specify the location of logback.xml.

Thanks for letting me know.

Patrik

**Comment by tony19** [comment]
You could keep one `assets/logback.xml` to configure `logback` to read config from another file location:

```xml
<configuration
  xmlns="https://tony19.github.io/logback-android/xml"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="https://tony19.github.io/logback-android/xml https://cdn.jsdelivr.net/gh/tony19/logback-android/logback.xsd"
>
  <includes>
    <include file="/path/to/your/logback.xml"/>
  </includes>
</configuration>
```
